back/app/services/user_service.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# This class exists to track and have in one place all operations
+# that write important user fields like email and confirmation.
+# It makes it easier to understand what and when can happen to user.
+#
+class UserService
+  class << self
+    def create_in_web_api(new_user, user_params, &)
+      new_user.assign_attributes(user_params)
+      yield
+      new_user.save(context: :form_submission) # `on: :create` and `on: :update` callbacks/validations are not called
+    end
+
+    def update_in_web_api(user, user_params, &)
+      user.assign_attributes(user_params)
+      yield
+      user.save(context: :form_submission) # `on: :create` and `on: :update` callbacks/validations are not called
+    end
+
+    def create_in_admin_api(user_params, confirm_user)
+      user = User.new(user_params)
+      user.locale ||= AppConfiguration.instance.settings('core', 'locales').first
+      user.confirm if confirm_user
+      user.save
+
+      user
+    end
```

Comment:
Wonder whether these would be better named `create` and `update` and then have a separate `AdminUserService` that inherits and overrides etc. But then the classes would be small now, if you then do the same for sso, importer etc

Reply:
(discussed on a meeting: for now, we keep it in one bigger file for better visibility)